src/klein/_isession.py

```diff
@@ -37,14 +32,96 @@ class TransactionEnded(Exception):
     """
 
 
+class SessionMechanism(Names):
```

Comment:
Moved class def up so that it's defined before it is used (as a type) later in this file.